### PR TITLE
Added joint_limits_interface - Direct port from ROS 1

### PR DESCRIPTION
**Pull request by ahcorde**
As discussed yesterday in the ros2_control WG, I'm adding this package originally ported by @ddengster

I still need to copy from @ddengster repository the `transmmision_interface` package.

@Karsten1987 @bmagyar 

Co-authored-by: ddengster <ed.fan@osrfoundation.org>

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by bmagyar** [comment]
@v-lopez @jordan-palacios could you guys confirm whether the license on this could also be updated?

**Comment by v-lopez** [comment]
Yes, no problem.

**Review by v-lopez — CHANGES_REQUESTED** [review]
I'd cleanup the test part before merging. Also made some minor suggestions.

**Comment by ahcorde** [comment]
Sorry for the bad conditions of the package. I was just copying a version from other repo that it was just functional.

I added 
 - common ROS 2 linters
 - Fixed test

File: joint_limits_interface/CMakeLists.txt
Comment (destogl) [inline joint_limits_interface/CMakeLists.txt:33]: Could you edit CMakeLists.txt to be somewhat in style with `hardware_interface/CMakeLists.txt`? This could increase readability.

File: joint_limits_interface/CMakeLists.txt
Comment (destogl) [inline joint_limits_interface/CMakeLists.txt:37]: as above

File: joint_limits_interface/CMakeLists.txt
Comment (destogl) [inline joint_limits_interface/CMakeLists.txt:50]: as above

File: joint_limits_interface/CMakeLists.txt
Comment (destogl) [inline joint_limits_interface/CMakeLists.txt:44]: as above

File: joint_limits_interface/README.md
Comment (destogl) [inline joint_limits_interface/README.md:20]: add space between the sentences

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp
Comment (destogl) [inline joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp:568]: It seems this code is never used. Do we still need it?

**Review by destogl — COMMENTED** [review]
First part of the review.

File: joint_limits_interface/package.xml
Comment (destogl) [inline joint_limits_interface/package.xml:1]: Why not use `format=3`

File: joint_limits_interface/package.xml
Comment (destogl) [inline joint_limits_interface/package.xml:8]: Add not-active maintainers as authors

File: joint_limits_interface/test/joint_limits_interface_test.cpp
Comment (destogl) [inline joint_limits_interface/test/joint_limits_interface_test.cpp:115]: why not to remove this?

File: joint_limits_interface/test/joint_limits_interface_test.cpp
Comment (destogl) [inline joint_limits_interface/test/joint_limits_interface_test.cpp:538]: remove?

**Review by destogl — CHANGES_REQUESTED** [review]
Also remove `.pyc` file.

I have to do it one more time

**Comment by destogl** [comment]
This looks good to me. I believe using the new concept from #121 this code is only temporarily. If not so there are few more issues. I list them here just for the "documentation":

* headers alpha sort
* There are some `TODO` comments in the code
* use of new JointHandle from #134

File: joint_limits_interface/package.xml
Comment (ahcorde) [reply joint_limits_interface/package.xml:8]: can you suggest me which are the authors and maintainer of this package?

File: joint_limits_interface/package.xml
Comment (ahcorde) [reply joint_limits_interface/package.xml:1]: The other packages are version 2, I can update this one.

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp
Comment (ahcorde) [reply joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp:568]: added a TODO

File: joint_limits_interface/package.xml
Comment (destogl) [reply joint_limits_interface/package.xml:8]: I think Enrique and Mathias are not active so much here, so I would move them in authors. @bmagyar is clearly maintainer :)

File: joint_limits_interface/test/joint_limits_interface_test.cpp
Comment (destogl) [inline joint_limits_interface/test/joint_limits_interface_test.cpp:4]: Is this potential issue in the `ros2_control` repository? Currently, we have everything Apache2.0 licensed. If so, can we discuss this with PAL to change the license?

**Review by destogl — APPROVED** [review]
It looks good!

File: joint_limits_interface/test/joint_limits_interface_test.cpp
Comment (v-lopez) [reply joint_limits_interface/test/joint_limits_interface_test.cpp:4]: If it is a problem we have no objections to change it to Apache2.0

Comment (destogl) [reply joint_limits_interface/test/joint_limits_interface_test.cpp:4]: @bmagyar

**Comment by guru-florida** [comment]
Unfortunately it looks like this merge didn't make it in time before the use of the new JointHandle from #134 took place (see also #155). I will see if I can resolve.

**Comment by destogl** [comment]
@ahcorde can we close this PR an open the new one wir changes from @guru-florida?


@guru-florida the changes in #134 are also kind of temporary since we will move everything to components in the future (when ResourceManager is finished - #164 and #147). Nevertheless, this restructuring you did is anyway needed since the logic behind is similar.

Maybe you have some idea how to integrate the limits into components. The example implementation of the PositionJoint is in #140

**Comment by guru-florida** [comment]
Thanks @destogl. I have created a new [PR181](https://github.com/ros-controls/ros2_control/pull/181).

**Comment by bmagyar** [comment]
Closing in favour of #181

**Pull request closed without merging**
